packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -1212,42 +1303,27 @@ class _ClassOrInstanceCountColumn extends ColumnData<Reference> {
     if (dataObject.name == MemoryController.libraryRootNode ||
         dataObject.name == MemoryController.classRootNode) return '';
 
-    if (dataObject.isExternals) {
-      if (dataObject.hasCount) return dataObject.count;
-
-      var count = 0;
-      for (ExternalReference externalRef in dataObject.children) {
-        count += externalRef.children.length;
-      }
-      return count;
-    } else if (dataObject.isExternal) {
+    if (dataObject.isExternal) {
       return dataObject.children.length;
-    } else if (dataObject.isFiltered) {
-      int sum = 0;
-      final FilteredReference filteredRef = dataObject;
-      for (final LibraryReference child in filteredRef.children) {
-        for (final HeapGraphClassLive liveClass in child.actualClasses) {
-          // Have the instances been realized (null implies no)
-          if (liveClass.instancesCount != null) {
-            sum += liveClass.instancesCount;
-          }
-        }
-      }
-
-      return sum;
     } else if (dataObject.isAnalysis && dataObject is AnalysisReference) {
       final AnalysisReference analysisReference = dataObject;
       final count = analysisReference.countNote;
       return count == null ? '' : count;
+    } else if (dataObject.isSnapshot && dataObject is SnapshotReference) {
+      final SnapshotReference snapshotRef = dataObject;
+      int count = 0;
+      for (final child in snapshotRef.children) {
+        count += _computeCount(child);
+      }
+      return count == null ? '--' : '$count';
```

Comment:
some cases have `'$count'` and some have `count`
Maybe make this a helper method and do the null handling outside the helper so you only have to do it once.

Reply:
String _displayCount(int count) => count == null ? '--' : '$count';